src/devices/nm-lldp-listener.c

```diff
@@ -267,6 +278,23 @@ _lldp_attr_set_uint32 (LldpAttrData *pdata, LldpAttrId attr_id, guint32 v_uint32
 	pdata->v_uint32 = v_uint32;
 }
 
+static void
+_lldp_attr_take_vardict (LldpAttrData *pdata, LldpAttrId attr_id, GVariant *variant)
+{
+
+	nm_assert (pdata);
+	nm_assert (_lldp_attr_id_to_type (attr_id) == LLDP_ATTR_TYPE_VARDICT);
```

Comment:
`nm_assert (variant && !g_variant_is_floating (variant));`.

However, it seems to me that all callers need to sinking the reference first. So, you don't actually safe a ref/sink/unref by the way this function works (and is used).
So, it would be more idiomatic that the function "takes" the variant by accepting a floating reference and sinking the reference on its own.

The name of the function is misleading. GVariants support to be floating, so API (that needs to keep the reference) generally consumes the provided floating reference by calling `g_variant_ref_sink()`. Calling the function `take_vardict` seems not right for something that happens commonly. Just call it `set_vardict()`.

The name also conflicts with `_lldp_attr_add_vardict()` below, but that other function also has an ill name (as it creates a different attribute type).

Reply:
Renamed the function to set_vardict() and changed it to take a reference (or sink the floating one).
Regarding naming, `add_vardict()` adds a variant to an attribute of type ARRAY_OF_VARDICTS, while `set_vardict()` sets a LLDP_ATTR_TYPE_VARDICT. Yeah, perhaps there are better names, but they don't seem wrong.